### DISCUSSION TITLE
Compare files on 2 machines

### DISCUSSION THREAD

**Discussion by petitours** [WinMerge/winmerge#1048] [Q&A]
Hello

Is it possible to run winmerge on 2 machines to obtain 2 list image files and be able to compare the 2 list image files on another machine ?

for example

Boris in Moscow runs winmerge on his pc and gets a list image file
Jack in Tahiti does the same on his pc and gets a list image file
Boris and Jack give their file to Peter who compares the files in Paris.

Possible?

thanks in advance

**Comment by sdottaka** [comment]
I may not have understood your question exactly, but WinMerge can output the comparison results of files or folders to a file, so you can output the comparison result file on two computers and compare the files further.

There are two types of comparison result files: report format and patch format.

To generate a comparison result file in report format, select Tools → Generate Report menu item.
To create a comparison result file in patch format, select Tools → Generate Patch menu item.

**Comment by petitours** [comment]
Thank for your answer

if I compare c: // TOTO with c: // TOTO I get a report file with no results and this is normal if I understood the comparison logic correctly.

I need to have a c: // TOTO inventory on each machine so that the 2 inventories can be compared elsewhere, without one of the machines being able to access the 2 compared machines at the same time.

In fact, I need the inventory before comparison, not the comparison directly.

**Comment by sdottaka** [comment]
What you want to do is probably similar to the following FeatureRequest, but unfortunately it is not possible with the current WinMerge.

https://github.com/WinMerge/winmerge/issues/618

However, you may be able to output a list of files to a text file at the command prompt on multiple computers and compare them as follows.

~~~.
dir /s c:\toto > c:\tmp\list1.txt
~~~

WinMerge 2.16.16 allows you to display the hash value of a file in a column of the folder comparison window, which is also output in the comparison report, as follows

https://github.com/WinMerge/winmerge/pull/996

You can use WinMerge to compare the same c:\TOTO folder and output a comparison report with the hash values, and then compare the report files to see if the files match exactly.

If you can't see the comparison results after comparing folders with the same path, click View → Show Identical items.

![image](https://user-images.githubusercontent.com/98126/142009029-60a2dc65-31de-4141-afab-befe8b9a0c68.png)
